src/Tasks/MigrateCommentParentsTask.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace SilverStripe\Comments\Tasks;
+
+use SilverStripe\Core\Config\Config;
+use SilverStripe\Dev\BuildTask;
+use SilverStripe\ORM\DB;
+
+/**
+ * Migrates all 3.x comment's BaseClass fields to the new ParentClass fields
+ *
+ * @package comments
+ */
+class MigrateCommentParentsTask extends BuildTask
+{
+    /**
+     * {@inheritDoc}
+     */
+    private static $segment = 'MigrateCommentParentsTask';
+
+    /**
+     * {@inheritDoc}
+     */
+    protected $title = 'Migrate Comment Parent classes from 3.x';
+
+    /**
+     * {@inheritDoc}
+     */
+    protected $description = 'Migrates all 3.x Comment BaseClass fields to the new ParentClass fields in 4.0';
+
+    /**
+     * {@inheritDoc}
+     */
+    public function run($request)
+    {
+        // Set the class names to fully qualified class names first
+        $remapping = Config::inst()->get('SilverStripe\\ORM\\DatabaseAdmin', 'classname_value_remapping');
```

Comment:
You need a check to see if BaseClass is a field on Comment; If it's not skip the upgrade.